### PR TITLE
<Date Picker /> - Today prop

### PR DESCRIPTION
**Pull request by poseneror**
### 🔦 Summary
As described here: https://jira.wixpress.com/browse/DSM-1628
A "today" prop has been added to the wix-style-react <Calenda /> and wix-ui-tpa <DatePicker />
The prop overrides the today indication, to handle timezone manipulations

* the indication had to be added conditionally, thus the extra if statement
...

### ✅ Checklist

- [x] 👨‍💻 API change is approved by the UI Infra Developers 
- 📚 Change is documented
  - [x] Story
  - [x] API description
- 🔬 Change is tested
  - [x] Visual test

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  Project Name: **wix-ui-tpa**

  You can find it under one of the following URLs:
  * https://bo.wix.com/pages/wix-ui-tpa.pr_6911/

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  Project Name: **wix-style-react**

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6911

File: packages/wix-style-react/src/Calendar/Calendar.js [hunk 1]
```diff
@@ -556,6 +563,9 @@ Calendar.propTypes = {
   */
   dateIndication: PropTypes.func,
 
+  /** Define today's date. The today indication is added automatically according to the user timezone but in some cases, we need the ability to add the today indication based on the business timezone. */
+  today: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(Date)]),
```
Comment (nitayneeman) [inline packages/wix-style-react/src/Calendar/Calendar.js:567]: Basically we're trying to get rid of the string parsing and to align with date-fns - which means, to support the native date object only.

How are you going to use it as a consumer?

**Review by nitayneeman — COMMENTED** [review]
Looks good! 👍

I wrote one question and also I think we should add an `exampleProp` for the playground to easily simulate this:
https://github.com/wix/wix-style-react/blob/master/packages/wix-ui-tpa/src/components/DatePicker/docs/index.story.tsx#L43

We can have something like:
```jsx
today: [
  {
    label: 'First day of month',
    value: new Date(new Date().getFullYear(), new Date().getMonth(), 1),
  },
  {
    label: 'Middle day of month',
    value: new Date(new Date().getFullYear(), new Date().getMonth(), 15),
  },
],
```

You can set the `new Date()` in a variable to make it clear although the above is alright as well IMO.

File: packages/wix-style-react/src/Calendar/Calendar.js (hunk 1 above)
Comment (poseneror) [reply packages/wix-style-react/src/Calendar/Calendar.js:567]: We are in fact using string dates, as Date objects are always in Local Timezone, which can be misleading. This is a known issue in javascript. We will send strings in the form of "2021-03-11T00:00:00"

**Review by nitayneeman — APPROVED** [review]
Although we plan to ditch the internal string parsing, we decided to align the API of this prop with the `value` prop - until we'll have a suggestion for a better API for both.

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

A deploy preview of your PR was successfully released 🖼.

Project Name: **wix-style-react**

You can find it under one of the following URLs:
* https://preview.wix-style-react.com/pr_6911

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

A deploy preview of your PR was successfully released 🖼.

Project Name: **wix-ui-tpa**

You can find it under one of the following URLs:
* https://bo.wix.com/pages/wix-ui-tpa.pr_6911/

**Pull request merged**